### DISCUSSION TITLE
Could you explain little bit more about the situation that vectorized modeling is challenging?

### DISCUSSION THREAD

**Discussion by happyych** [fumitoh/modelx#101] [Q&A] [answered]
Could you explain little bit more about the situation that vectorized modeling is challenging?

I read the article about export feature impressively. 
https://modelx.io/blog/2023/08/19/enhanced-speed-for-exported-lifelib-models/

While reading the article, I cannot figure out the situation you mentioned as the limitation of the vectorized modeling,
**" modeling requirements where projections from both the current and future points in time are needed can be challenging for vectorized models"**

Could you explain the above situation in more detail? Telling me other cases that are hard to modeling in vectorized model would be
also appreciated.

Thanks everytime!

**Comment by fumitoh** [comment, marked as answer]
An example of the scenario mentioned in the blog post is where you need to perform nested projections for future points in time to calculate reserves at those points due to reserving requirements in some countries. Vectorized models may be able to implement this logic, but the logic would be very complex.

Another case where it's hard for vectorized models to implement is retrospective projections for existing policies from their issue dates. This is the case in US-GAAP LDTI and FAS97, where you need to consider past experience up to the current date in combination with future assumptions to calculate reserves. Since the issue dates of existing policies are scattered, it would be hard for a vectorized model to apply the same calculation logic to vectors representing the entire set of existing policies.

**Reply by happyych** [reply]
Thank you for your kind explanation!
Have a nice day
